p2p/protocol/head/head.go

```diff
@@ -28,7 +42,6 @@ func (p *Publisher) Serve(host host.Host, topic string) error {
 		return err
 	}
 
-	p.server = &http.Server{Handler: http.Handler(p)}
```

Comment:
Yeah this is a bit subtle. In general we shouldn't be mutating the object that can be shared across goroutines without a mutex/other-sync-primitive. And we can do this part in the New func.

Reply:
Thanks for making this change! 👍